### ISSUE TITLE
Search context dropdown is not accessible/violates ARIA role="menu" spec

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#20234]
The search context dropdown uses a Bootstrap dropdown, which is a generic dropdown system supporting arbitrary content.

It is also marked up with ARIA's `role="menu"`, meaning it is exposed to screen readers as an ARIA menu widget and must follow the specification for `role="menu"` widgets. This is a subset of what dropdowns can be used for, see:

> The WAI ARIA standard defines an actual role="menu" widget, but this is specific to application-like menus which trigger actions or functions. ARIA menus can only contain menu items, checkbox menu items, radio button menu items, radio button groups, and sub-menus.

https://getbootstrap.com/docs/5.0/components/dropdowns/#accessibility

And directly from the [ARIA spec](https://www.w3.org/TR/wai-aria-practices/#wai-aria-roles-states-and-properties-13):

> - A menu is a container of items that represent choices. The element serving as the menu has a role of either `menu` or `menubar`.
> - The items contained in a menu are child elements of the containing menu or menubar and have any of the following roles:
>   - `menuitem`
>   - `menuitemcheckbox`
>   - `menuitemradio`

The search context dropdown though contains more than just those items: It has a search box, and a "Reset" button that is not a `menuitem`. Additionally, the `menuitem`s are not child elements of the `menu` element.

Furthermore, the [success criteria for `menu`s](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus) and [testing procedures](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus) are failed by this "menu". E.g. one of the criteria is that when opening a menu with the keyboard, the first menu item must be focused. Instead, the search box (which is not a menu item) is focused.

![image](https://user-images.githubusercontent.com/10532611/115462866-73801780-a22b-11eb-8395-4a8e48c986e8.png)

I would suggest that the button is instead marked up as a `aria-haspopup="dialog"` and the container for the popup content as a `role="dialog"` instead.

> Authors MUST ensure that the role of the element that serves as the container for the popup content is menu, listbox, tree, grid, or dialog, and that the value of aria-haspopup matches the role of the popup container.

https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup

Interestingly the button that opens the menu is currently _not_ marked up as a `menubutton` as would be needed if this was a menu.

**Comment by camdencheek** [comment]
Closing since we no longer use the bootstrap dropdown
